### DISCUSSION TITLE
OpenMVS vs texrecon

### DISCUSSION THREAD

**Discussion by satyajitghana** [cdcseacave/openMVS#957] [Ideas]
I tried openMVS texturing and also texrecon https://github.com/nmoehrle/mvs-texturing

It seems texrecon is much better in terms of quality of texture generated. openMVS creates a large patch of texture by default and tries to pack all the views in that. in texrecon each view is projected individually and separate texture images for each view is saved. which does generate a large number of file is views is more, but the final model looks much better.

It would be nice if texrecon is merged with openMVS texturing pipeline.

**Comment by cdcseacave** [comment]
not sure what you mean exactly, can you pls share the same model (preferably starting from the same SfM solution) reconstructed in both OpenMVS and texrecon?

**Comment by ghost** [comment]
# Ammonite | 102 Images 6000x4000
    TextureMesh -i scene_dense_mesh.mvs --resolution-level 0 --outlier-threshold 0.078 --patch-packing-heuristic 0
      --decimate 0.05 --virtual-face-images 3
        Time: 6m 30s
        Atlases: 1
      --decimate 0.01 --virtual-face-images 3
        Time: 1m 52s
        Atlases: 1
    texrecon
      Decimation 1
        Time: 7m 55s
        Atlases: 55
      Decimation 0.05
        Time: 40s
        Atlases: 5
TextureMesh 0.05 vf3
![0 05_vf300](https://user-images.githubusercontent.com/108303308/225808281-b7bdb538-77e2-434b-a59d-41c37cc7b6cc.png)
texrecon 0.05
![0 05_texrecon00](https://user-images.githubusercontent.com/108303308/225808289-23ef80e6-da4b-43b3-a2d5-14fd0814bf0d.png)
texrecon 1
![snapshot01](https://user-images.githubusercontent.com/108303308/225810897-922f77ce-2877-44e1-890d-561ef0853c73.png)

**Reply by satyajitghana** [reply]
> # Ammonite | 102 Images 6000x4000
> ```
> TextureMesh -i scene_dense_mesh.mvs --resolution-level 0 --outlier-threshold 0.078 --patch-packing-heuristic 0
>   --decimate 0.05 --virtual-face-images 3
>     Time: 6m 30s
>     Atlases: 1
>   --decimate 0.01 --virtual-face-images 3
>     Time: 1m 52s
>     Atlases: 1
> texrecon
>   Decimation 1
>     Time: 7m 55s
>     Atlases: 55
>   Decimation 0.05
>     Time: 40s
>     Atlases: 5
> ```
> 
> TextureMesh 0.05 vf3 ![0 05_vf300](https://user-images.githubusercontent.com/108303308/225808281-b7bdb538-77e2-434b-a59d-41c37cc7b6cc.png) texrecon 0.05 ![0 05_texrecon00](https://user-images.githubusercontent.com/108303308/225808289-23ef80e6-da4b-43b3-a2d5-14fd0814bf0d.png) texrecon 1 ![snapshot01](https://user-images.githubusercontent.com/108303308/225810897-922f77ce-2877-44e1-890d-561ef0853c73.png)


look at the smudges in TextureMesh

<img width="1719" alt="Screenshot 2023-03-17 at 10 28 27 AM" src="https://user-images.githubusercontent.com/8092481/225824258-a090cc10-682e-439e-9f9f-66db490bc3c0.png">

These are much better in texrecon

**Comment by cdcseacave** [comment]
this is a great comparison, thank, however why do you use the virtual faces feature? it is work in progress; or at least compare also to the defualt (mature) algorithm, to be fair, right?

**Reply by ghost** [reply]
True. I used virtual faces to give TextureMesh its best shot at quality.
#
    TextureMesh
      --decimate 0.05
        Time: 2m 51s
        Atlases: 1
![TextureMesh_0 0500](https://user-images.githubusercontent.com/108303308/225972048-abb7e64f-924e-42c0-8147-1a81704fbbf5.png)
#
    TextureMesh
      --decimate 0.05 --cost-smoothness-ratio 1
        Time: 3m 12s
        Atlases: 1
![TextureMesh_0 05_cs100](https://user-images.githubusercontent.com/108303308/225974397-10103f0a-6d0d-4dc1-864d-b8dbc8e9245f.png)
#
    TextureMesh
      --decimate 0.05 --cost-smoothness-ratio 1 --outlier-threshold 0.0599999987
        Time: 3m 14s
        Atlases: 1
![snapshot00](https://user-images.githubusercontent.com/108303308/225975838-a7077809-06e2-4728-b47f-6b4719726d13.png)
#
    TextureMesh
      --decimate 0.05 --virtual-face-images 3 --outlier-threshold 0.0599999987
        Time: 5m 28s
        Atlases: 1
![uwu00](https://user-images.githubusercontent.com/108303308/225978943-8085aaaa-d583-4794-87b5-a3e5308c6b1f.png)
#
    TextureMesh
      --decimate 0.05 --virtual-face-images 3 --cost-smoothness-ratio 1 --outlier-threshold 0.0599999987
        Time: 5m 11s
        Atlases: 1
![owo00](https://user-images.githubusercontent.com/108303308/225980825-24439600-7c55-4545-89db-113854af835a.png)

**Comment by cdcseacave** [comment]
very interesting, it seems that each param has quite some influence on the final result when it should not, like the `--outlier-threshold` for ex
next it would be interesting in identifying the weak link, if it is only one,
anybody willing to help with this?
